cpp/src/io/json/reader_impl.cu

```diff
@@ -464,47 +475,71 @@ void reader::impl::set_column_names(device_span<uint64_t const> rec_starts,
   }
 }
 
-void reader::impl::set_data_types(device_span<uint64_t const> rec_starts,
-                                  rmm::cuda_stream_view stream)
+std::vector<data_type> reader::impl::parse_data_types(
```

Comment:
the diff is very confusing here. Did not rename the function, but added `parse_data_types` to handle the deprecated API that takes strings. This will make the future removal cleaner.